### PR TITLE
Add input tags component

### PR DESCRIPTION
**Pull request by MathieuDebit**
image

## Usage
See [its story](https://github.com/getstation/theme/pull/41/files#diff-216ef3016a48a721d08c0083055f60f4R17) and [exported types](https://github.com/getstation/theme/pull/41/files#diff-539bf2a6e582fc26812af1be629c20caR131).

```typescript
<InputTags
          items={} // type InputTagsItem[] (Array of {id: string, name: string, selected: boolean, picture: string})
          onUpdateTags={} // takes type InputTags as params
        />
```

## TODO
- [ ] Changelog
- [x] Remove tags props and get it from items selected
- [x] Update stories
- [ ] fix size of input
- [ ] fix color of checkmark
- [x] fix `ContainerOpen` position with multiple line tags
- [ ] clean code and props

### CODE REVIEW

**Comment by alexstrat** [comment]
Feels like the lint is unhappy! :)

**Comment by MathieuDebit** [comment]
Yes I'm on it!

**Comment by alexstrat** [comment]
Was published yesterday as `0.14.1-rc2`.

@MathieuDebit the next version is `0.15.0` (feature addition), so  `0.15.0-rc`s would be more adapted :)

File: lib/components/InputTags/index.tsx [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+        return <TagsInput
+            renderInput={this.autocompleteRenderInput}
+            renderTag={(renderTagsProps: any) => renderTag(this.props, renderTagsProps)}
+            value={this.props.tags}
+            onChange={this.props.onUpdateTags}
+        />
+    }
+}
+
+export const InputTags = injectSheet(styles as IgnoreJSSNested<typeof styles>)(InputTagsImpl);
```
Comment (alexstrat) [inline lib/components/InputTags/index.tsx]: please export as `React.Components< OwnProps >`

File: lib/components/InputTags/index.tsx [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+        marginRight: 6,
+    },
+    renderTagContent: {
+    },
+    renderTagRemove: {
+    },
+};
+
+interface OwnProps {
+    items: InputTagsItem[],
```
Comment (alexstrat) [inline lib/components/InputTags/index.tsx]: What is the difference between items and tags?

File: lib/components/InputTags/index.tsx [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+    tags: InputTags,
+    onUpdateTags: (tags: InputTags) => any,
+}
+
+export interface InputTagsItem {
```
Comment (alexstrat) [inline lib/components/InputTags/index.tsx]: I think an `id` would be useful here.

File: lib/components/InputTags/index.tsx [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+                onSuggestionsFetchRequested={() => {}}
+                theme={theme}
+            />
+        )
+    };
+
+    render () {
+        function renderTag(props: Props, renderTagsProps: any) {
+            const { classes, items } = props;
+            const { tag, key, disabled, onRemove, className, classNameRemove, getTagDisplayValue, ...other } = renderTagsProps;
```
Comment (alexstrat) [inline lib/components/InputTags/index.tsx]: What would the key here? are you sure there is one?

File: lib/components/InputTags/index.tsx (hunk 2 above)
Comment (MathieuDebit) [reply lib/components/InputTags/index.tsx]: Items are all the elements to display in the list whereas tags are the selected ones. I removed the tags prop to only have items that have a `selected` boolean

**Comment by MathieuDebit** [comment]
`0.15.0-rc1` published with allmost all fixes

File: lib/components/InputTags/index.tsx [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+  },
+  renderTagContent: {
+  },
+  renderTagRemove: {
+  },
+};
+
+interface OwnProps {
+  items: InputTagsItem[],
+  onUpdateTags: (tags: InputTags) => any,
```
Comment (alexstrat) [inline lib/components/InputTags/index.tsx]: onUpdateTags: (tags: InputTags) => void

File: package.json [hunk 6]
```diff
@@ -67,6 +74,7 @@
     "@types/storybook__react": "^3.0.8",
     "babel-core": "^6.26.3",
     "babel-runtime": "^6.26.0",
+    "css-loader": "^3.2.0",
```
Comment (alexstrat) [inline package.json]: Why is that necessary?

File: lib/components/InputTags/index.tsx [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+}
+
+export interface InputTagsItem {
+  id: string,
+  name: string,
+  selected: boolean,
+  picture: string,
+}
+
+export type InputTags = String[];
```
Comment (alexstrat) [inline lib/components/InputTags/index.tsx]: `InputTags` is also exported as a constant which typing does not correspond to this one

File: lib/components/InputTags/index.tsx (hunk 5 above)
Comment (alexstrat) [reply lib/components/InputTags/index.tsx]: It's not clear what to expect in `tags`. It looks like it's a `string[]` but what are the `string`? Ideally `ids`. Worth adding a JSDoc

```
/**
* It will be called with the `id`s of the `items` that are selected once it is updated..
*/
 onUpdateTags: (tags: InputTags) => void,
```
